src/libudev/libudev-monitor.c

```diff
@@ -702,6 +703,19 @@ _public_ struct udev_device *udev_monitor_receive_device(struct udev_monitor *ud
         return udev_device;
 }
 
+int udev_monitor_receive_sd_device(struct udev_monitor *udev_monitor, sd_device **ret) {
+        _cleanup_(udev_device_unrefp) struct udev_device *udev_device = NULL;
+
+        assert(ret);
+
+        udev_device = udev_monitor_receive_device(udev_monitor);
+        if (!udev_device)
+                return -errno;
+
+        *ret = sd_device_ref(udev_device->device);
```

Comment:
At some point we probably want to do the reverse, and make `udev_monitor_receive_device` a wrapper around `udev_monitor_receive_sd_device`.

Reply:
Yeah. It was done in the previous PR, but this PR does not include cleanups for libudev...